### ISSUE TITLE
Index was outside the bounds of the array

### ISSUE THREAD

**Issue by Erikov-K** [iRon7/ConvertFrom-SourceTable#3]
Hi!

Thank you for script 'ConvertFrom-SourceTable.ps1'
Great work! :-)

In my project I use [StrictMode ](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/set-strictmode?view=powershell-7)Version 3.0 and I see that function 'ConvertFrom-SourceTable' throw 'Index was outside the bounds of the array' exception.

Simple example below:
```
PS C:\> Set-StrictMode -Version 2
PS C:\> . .\ConvertFrom-SourceTable.ps1
PS C:\> ConvertFrom-SourceTable '
>> Name   Age
>> ----   ----
>> John   18
>> Sam    22
>> ' -ErrorAction stop

Name Age
---- ---
John 18
Sam  22


PS C:\> Set-StrictMode -Version 3
PS C:\> ConvertFrom-SourceTable '
>> Name   Age
>> ----   ----
>> John   18
>> Sam    22
>> ' -ErrorAction stop
Foreach-Object : Index was outside the bounds of the array.
At C:\ConvertFrom-SourceTable.ps1:377 char:90
+ ... Skip $DataIndex | Where-Object {$_.Trim()} | Foreach-Object {Mask $_}
+                                                  ~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (:) [ForEach-Object], IndexOutOfRangeException
    + FullyQualifiedErrorId : System.IndexOutOfRangeException,Microsoft.PowerShell.Commands.ForEachObjectCommand

PS C:\>

```

**Comment by iRon7** [comment]
Thanks for the information, I wasn't even aware that is a new `Set-StrictMode -Version 3` 😳
I will need to investigate this.

**Comment by iRon7** [comment]
Apparently I was implicitly relying on the behavior that an index outside the bounds of the array returns `$Null` with statements like `$Array[$i] -notmatch '\S'` (vs. `$Array[$i] -match '\s'`).
Anyways, it should be fixed now in version `0.3.15`
